### DISCUSSION TITLE
Loading program memory from txt file

### DISCUSSION THREAD

**Discussion by SuperHerc** [hneemann/Digital#1370] [Q&A]
I am trying to load my CPUs program memory before startup using an external file (a txt file) and I cant seem to figure out how it loads in the data. When I give it data in hex, binary, decimal, etc it always overflows the memory location into even though the space should be sufficient. For example my memory can hold 2 bytes in it but when I write 4 hex characters for those two bytes and start the program, the memory is filled all the way through addresses 0-5 and the number is way larger than it should be. I cant seem to find documentation on to properly load it in. Any help is appreciated.

<img width="415" alt="Screenshot 2024-12-13 at 9 02 15 AM" src="https://github.com/user-attachments/assets/8607f379-67be-428a-a9d7-a2d9f0429a11" />

<img width="493" alt="Screenshot 2024-12-13 at 9 02 58 AM" src="https://github.com/user-attachments/assets/5de19d2b-00c3-413f-822f-e8161518453d" />

I want to load 0x8001 into address 0 of memory but I don't get how it is loading in.

**Comment by Slicudis** [comment]
The file must have the .hex format to load Logisim hex files. To be honest, you should directly just use raw and edit the file in a hex editor

**Reply by fredcwbr** [reply]
Howdy HNeeman,
First of all , great work ,
I do understand the binary load that is going on, and it fulfill my needs,  
but it would be nice if Digital could load intel Hex files to a RAM/ROM like component.

Anyway .. as i said before ., awesome work , very grateful for it., 
Thanks very much!!

**Reply by hneemann** [reply]
Intel-Hex-Files are supported. 
If the filename ends with `.hex`  a Intel-Hex file importer is used to load the file.

**Reply by xotmatrix** [reply]
Digital can load Intex .hex files which have a precise format using a length, address, record type, data, and checksum per line.
```
:10000000000102030405060708090A0B0C0D0E0F78
:00000001FF
```

And it can load a simpler "raw" format with one value per line.
```
v2.0 raw
00
01
02
03
04
05
06
07
08
09
0A
0B
0C
0D
0E
0F
```

**Comment by fredcwbr** [comment]
This is great!!
Sorry i didn't get from the docs., 

Digital Rocks... 
Thanks for this great work.
